mono/metadata/icall-def.h

```diff
@@ -896,6 +896,13 @@ ICALL(ILOCK_20, "Increment(int&)", ves_icall_System_Threading_Interlocked_Increm
 ICALL(ILOCK_21, "Increment(long&)", ves_icall_System_Threading_Interlocked_Increment_Long)
 ICALL(ILOCK_22, "Read(long&)", ves_icall_System_Threading_Interlocked_Read_Long)
 
+// A subset of System.Diagnostics.PerformanceCounter usable in mscorlib.
+// Note that System.Diagnostics.PerformanceCounter is mostly native,
+// but the metadata and small amount of implementation is in System.dll.
+ICALL_TYPE(THREADP_PERFCTR, "System.Threading.InternalPerformanceCounter", THREADP_PERFCTR_1)
+ICALL(THREADP_PERFCTR_1, "GetImpl", mono_perfcounter_get_impl)
+ICALL(THREADP_PERFCTR_2, "UpdateValue", mono_perfcounter_update_value)
```

Comment:
Why is this needed? We already expose `mono_perfcounter_update_value` as `System.Diagnostics.PerformanceCounter:UpdateValue`.

As an aside: multiple icalls to the same C function is sometimes asking for trouble: people (e.g. me) sometimes forget to update both managed call sites when changing the native implementation because it's not obvious that there are multiple managed declarations.

Reply:
Unclear if to wrap in ifdef -- the managed code makes the calls unconditionally.